awscli/customizations/commands.py

```diff
@@ -309,6 +310,14 @@ def _raise_usage_error(self):
         ) % lineage
         raise ParamValidationError(error_msg)
 
+    def _add_customization_to_user_agent(self):
+        if 'command' in self._session.user_agent_extra:
```

Comment:
So I'm a little hesitant on doing this if check because:

1) The command could technically have `command` in its name causing us to inappropriate append the last element in the lineage to it. I could not find an example of where this is a problem now but does not mean it could be a problem in the future.

2) The presence of `command` does not necessarily mean we are building up the command component of the user agent as some other logic could have come along and appended to the user agent.

I'm thinking it may make sense to change this to a regex that searches for `command/` and replaces its value component with the current command lineage (and not just appending the last element of the lineage). Then if the regex does not find anything then we just fall to the logic in the if statement.

Reply:
I think it makes sense to look for `command` + `/` but I'd prefer not to user regexps substitutions because now user_agent is sort of immutable because we don't change anything in the current one just add to the end

Reply:
That's fair on the regex idea.